experimental/Rings/WeierstrassPreparationTheorem.jl

```diff
@@ -0,0 +1,239 @@
+##############################################################
+#
+# Conversion from Singular (weierstr.lib)
+#
+#############################################################
+
+export weierstrass_preparation_theorem
+
+export weierstrass_division_theorem
+
+
+"""
+    order_of_generality(f::MPolyElem, n::Int)
+
+Compute for a given polynomial `f` an integer `d` such that `f` is `x_n`-general of order `d` 
+with `x_n` the `n`-th variable of the polynomial ring of `f`.
+"""
+function order_of_generality(f::MPolyElem, n::Int)
+	 
+	vect = fill(parent(f)(0), nvars(parent(f))) 
+	vect[n] = gen(parent(f),n) 
+	f = evaluate(f, vect) 
+	f != 0 || return false, -1
+	return true, degree(leading_monomial(f, ordering=neglex(gens(parent(f)))), n)
+end
+
+"""
+    jet(f::MPolyElem, degree_f::Int)
+
+Given a polynomial `f` and an integer degree `degree_f` , this function returns the polynomial,
+which contains only the monomials of `f`, which total degree is lower than or equal to `degree_f`.
+"""
+function jet(f::MPolyElem, degree_f::Int) 
+  i = length(f)
+  while i >= 1
+    if total_degree(term(f,i)) > degree_f
+      f = f - term(f,i)
+    end
+    i = i - 1
+  end 
+	return f
+end
+
+
+"""
+    jet(f::MPolyElem, degree_f::Int, vecw::Vector{Int})
+
+Given a polynomial `f`, an integer degree `degree_f` and an integer vector `vecw`, which has to have 
+the same length as length(parent(f)), this function returns a polynomial, which only contains 
+monomial, whose exponents taken as a vector multiplied (scalar product) with the given 
+weight-vector `intw` are lower than or equal to `degree_f`.
```

Comment:
```suggestion
weight-vector `vecw` are lower than or equal to `degree_f`.
```
(Although the `vec` is somewhat redundant, so perhaps just call it `w` or `weights`?)